### PR TITLE
Add Bitbucket Cloud Support as source

### PR DESCRIPTION
**Pull request by Doctor-X**
In my project, we needed to source the migrations from Bitbucket Cloud. 
Implemented the source Driver interface for Bitbucket Cloud, let me know if this can be merged into the codebase.

### CODE REVIEW

File: source/bitbucket/bitbucket.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	if len(pe) < 1 {
+		return nil, ErrInvalidRepo
+	}
+	bi.config.Repo = pe[0]
+	if len(pe) > 1 {
+		bi.config.Path = strings.Join(pe[1:], "/")
+	}
+	bi.config.Ref = u.Fragment
+
+	if err := bi.readDirectory(); err != nil {
```
Comment (dhui) [inline source/bitbucket/bitbucket.go:78]: Call `WithInstance()` so there's a single path to instantiate a `Bitbucket` struct.

**Review by dhui — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: source/bitbucket/bitbucket.go (hunk 1 above)
Comment (Doctor-X) [reply source/bitbucket/bitbucket.go]: done!!

File: source/bitbucket/bitbucket_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+func Test(t *testing.T) {
+	if len(BitbucketTestSecret) == 0 {
+		t.Skip("test requires .bitbucket_test_secrets")
+	}
+
+	b := &Bitbucket{}
+
+	d, err := b.Open("bitbucket://" + BitbucketTestSecret + "@abhishekbipp/test-migration/migrations/test#master")
```
[the remark this answers is not in the document]
Comment (Doctor-X) [reply source/bitbucket/bitbucket_test.go:27]: It's a public repo. 
https://bitbucket.org/abhishekbipp/test-migration/src/master/migrations/test/

**Review by dhui — APPROVED** [review]

**Pull request merged**